frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,255 @@
+import './_restore-pvc-modal.scss';
+
+import * as React from 'react';
+
+import { Form, FormGroup, Grid, GridItem, TextInput, Alert } from '@patternfly/react-core';
+import {
+  HandlePromiseProps,
+  ResourceIcon,
+  withHandlePromise,
+  validate,
+  resourceObjPath,
+  history,
+  RequestSizeInput,
+  Timestamp,
+  LoadingInline,
+} from '@console/internal/components/utils';
+import {
+  K8sResourceKind,
+  k8sCreate,
+  referenceFor,
+  VolumeSnapshotKind,
+  StorageClassResourceKind,
+  k8sGet,
+  PersistentVolumeClaimKind,
+} from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import {
+  NamespaceModel,
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+} from '@console/internal/models';
+import { getName, getNamespace, Status } from '@console/shared';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+
+const dropdownUnits = {
+  Mi: 'MiB',
+  Gi: 'GiB',
+  Ti: 'TiB',
+};
+
+export const RestorePVCModal = withHandlePromise((props: RestorePVCModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress, handlePromise } = props;
+  const [restorePVCName, setPVCName] = React.useState(`${getName(resource) || 'pvc'}-restore`);
+  const defaultSize: string[] = resource?.status?.restoreSize
+    ? validate.split(resource?.status?.restoreSize)
+    : null;
+  const pvcRequestedSize = resource?.status?.restoreSize
```

Comment:
Is PVC requested size always equal to restore size?

Reply:
yes, ideally the requested size of PVC should be always equal or greater than restore the size of the snapshot. Just to show the initial value of possible requested size, we can consider `restoreSize`